ci-operator/step-registry/assisted/baremetal/test/assisted-baremetal-test-commands.sh

```diff
@@ -26,32 +25,71 @@ trap collect_artifacts EXIT TERM
+test_env_file="test-env"
+
+echo "${TEST_LIST:-""}" > "${ARTIFACT_DIR}/${test_list_file}"
+echo "${TEST_SKIPS:-""}" > "${ARTIFACT_DIR}/${test_skips_file}"
+cat << EOF > "${ARTIFACT_DIR}/${test_env_file}"
+openshift_tests_image="${OPENSHIFT_TESTS_IMAGE}"
+test_type="${TEST_TYPE:-"list"}"
+test_suite="${TEST_SUITE:-"openshift/conformance/parallel"}"
+test_provider="${TEST_PROVIDER:-"baremetal"}"
+test_list_file="/tmp/${test_list_file}"
+test_skips_file="/tmp/${test_skips_file}"
+EOF
+
+timeout --kill-after 10m 120m scp "${SSHOPTS[@]}"   \
+    "${ARTIFACT_DIR}/${test_list_file}"             \
+    "${ARTIFACT_DIR}/${test_skips_file}"            \
+    "${ARTIFACT_DIR}/${test_env_file}"              \
+    "root@${IP}:/tmp"
 
 echo "### Running tests"
-timeout --kill-after 10m 120m ssh "${SSHOPTS[@]}" "root@${IP}" bash -s "${OPENSHIFT_TESTS_IMAGE}" << "EOF"
+timeout --kill-after 10m 120m ssh "${SSHOPTS[@]}" "root@${IP}" "bash -s" << "EOF"
     set -x
 
-    function get_test_list() {
-        podman run --network host --rm -i -e KUBECONFIG=/tmp/kubeconfig -v ${kubeconfig}:/tmp/kubeconfig $1 \
-            openshift-tests run "openshift/conformance/parallel" --dry-run | \
-            grep -Ff /tmp/test-list
+    source /tmp/test-env
+
+    function get_baremetal_test_list() {
+        podman run --network host --rm -i \
+            -e KUBECONFIG=/tmp/kubeconfig -v "${KUBECONFIG}:/tmp/kubeconfig" "${openshift_tests_image}" \
+            openshift-tests run "${test_type}" \
+            --dry-run \
+            --provider "{\"type\": \"${test_suite}\"}"
     }
 
     function run_tests() {
-        podman run --network host --rm -i -v /tmp:/tmp -e KUBECONFIG=/tmp/kubeconfig -v ${kubeconfig}:/tmp/kubeconfig $1 \
-            openshift-tests run -o /tmp/artifacts/e2e_${name}.log --junit-dir /tmp/artifacts/reports -f -
+        podman run --network host --rm -i -v /tmp:/tmp -e KUBECONFIG=/tmp/kubeconfig -v "${KUBECONFIG}:/tmp/kubeconfig" "${openshift_tests_image}" \
+            openshift-tests run -o "/tmp/artifacts/e2e_${name}.log" --junit-dir /tmp/artifacts/reports --file "${test_list_file}"
     }
     
     # prepending each printed line with a timestamp
     exec > >(awk '{ print strftime("[%Y-%m-%d %H:%M:%S]"), $0 }') 2>&1
 
-    for kubeconfig in $(find ${KUBECONFIG} -type f); do
-        export KUBECONFIG=${kubeconfig}
-        name=$(basename ${kubeconfig})
-
-        stderr=$( { get_test_list $1 | run_tests $1; } 2>&1 )
+    for kubeconfig in $(find "${KUBECONFIG}" -type f); do
+        export KUBECONFIG="${kubeconfig}"
+        name=$(basename "${kubeconfig}")
+
+        case ${test_type} in
+            suite)
+                get_baremetal_test_list > "${test_list_file}"
+                ;;
+            list)
+                ;;
+            *)
+                echo >&2 "Unsupported TEST_TYPE: ${test_type}"
+                exit 1
+                ;;
+        esac
+
+        mv "${test_list_file}" "${test_list_file}.bak"
```

Comment:
Why ?

Reply:
because the pattern `cat <file> > <file>` corrupts the file